### PR TITLE
OSDOCS-4933: Add migration to OpenShift SDN

### PR DESCRIPTION
**Pull request by jboxman-rh**
- https://issues.redhat.com/browse/OSDOCS-4933

This procedure is similar to the existing roll back procedure, which can be repurposed for this with updates.

Previews:

- [Migrating to the OpenShift SDN network plugin](https://56592--docspreview.netlify.app/openshift-enterprise/latest/networking/openshift_sdn/migrate-to-openshift-sdn.html)
- [Rolling back to the OpenShift SDN network provider](https://56592--docspreview.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/rollback-to-openshift-sdn.html): Confirm conditional includes import block only here, not in prior link
- [Migrating from the OpenShift SDN network plugin](https://56592--docspreview.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.html#how-the-migration-process-works_migrate-from-openshift-sdn): Confirm this is to OVN-Kubernetes, from OpenShift SDN, as before but using attributes to share content with first link, while maintaining coherency
- [Roll back to the OVN-Kubernetes network plugin](https://56592--docspreview.netlify.app/openshift-enterprise/latest/networking/openshift_sdn/rollback-to-ovn-kubernetes.html)

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by jboxman-rh** [comment]
@pliurh, this PR mostly abstracts the existing roll back procedure and reuses it as part of a new topic, under the OpenShift SDN section, about migrating _to_ the OpenShift SDN plugin. I also reuse the migration process table. See the PR description for preview links.

Let me know what you think; Thanks!

**Comment by jboxman-rh** [comment]
@weliang1, can you review this documentation add for migration to the OpenShift SDN network plugin? It's mostly a restructure of the existing rollback to OpenShift SDN documentation. Thanks!

**Comment by jboxman-rh** [comment]
@pliurh @weliang1 ping

**Comment by weliang1** [comment]
@jboxman-rh 

`For your reference:  https://github.com/openshift/release/blob/master/ci-operator/step-registry/ovn/sdn-migration-rollback/ovn-sdn-migration-rollback-commands.sh

1. Add below step before step 2 in https://56592--docspreview.netlify.app/openshift-enterprise/latest/networking/openshift_sdn/migrate-to-openshift-sdn.html
# Reset the spec.migration before we can set it to other value
oc patch Network.operator.openshift.io cluster --type='merge' --patch '{"spec":{"migration":null}}'

2. For step 6 in https://56592--docspreview.netlify.app/openshift-enterprise/latest/networking/openshift_sdn/migrate-to-openshift-sdn.html, if ssh access is not available, we can try below steps:
# Reboot all the nodes
readarray -t POD_NODES <<< "$(oc get pod -n openshift-machine-config-operator -o wide| grep daemon|awk '{print $1" "$7}')"
for i in "${POD_NODES[@]}"
do
    read -r POD NODE <<< "$i"
    until oc rsh -n openshift-machine-config-operator "$POD" chroot /rootfs shutdown -r +1; do echo "cannot reboot node $NODE, retry"&&sleep 3; done
done 
`

**Comment by pliurh** [comment]
We have rollback instructions for sdn->ovn-k migration. Shall we also rollback instructions for ovn-k->sdn migration?

**Comment by jboxman-rh** [comment]
> We have rollback instructions for sdn->ovn-k migration. Shall we also rollback instructions for ovn-k->sdn migration?

We could. I wasn't sure how many users would actually want to migrate back to OpenShift SDN in the first place. I assumed anyone doing so would really want to be there.

**Comment by jboxman-rh** [comment]
@pliurh, the rollback procedure would be identical to the procedure to migration to OVN-Kubernetes?

**Comment by pliurh** [comment]
> @pliurh, the rollback procedure would be identical to the procedure to migration to OVN-Kubernetes?

Yes.

**Comment by jboxman-rh** [comment]
@pliurh, this adds that procedure: https://56592--docspreview.netlify.app/openshift-enterprise/latest/networking/openshift_sdn/rollback-to-ovn-kubernetes.html

File: modules/nw-ovn-kubernetes-migration.adoc [hunk 1]
```diff
@@ -277,7 +278,26 @@ Waiting for daemon set "multus" rollout to finish: 5 of 6 updated pods are avail
 daemon set "multus" successfully rolled out
 ----
 
-. To complete the migration, reboot each node in your cluster. For example, you can use a bash script similar to the following example. The script assumes that you can connect to each host by using `ssh` and that you have configured `sudo` to not prompt for a password.
+. To complete changing the network plugin, reboot each node in your cluster. You can reboot the nodes in your cluster with either of the following approaches:
+
+** With `oc rsh`, you can use a bash script similar to the following.
```
Comment (adellape) [inline modules/nw-ovn-kubernetes-migration.adoc:283]: s/following./following:/

File: modules/nw-ovn-kubernetes-migration.adoc [hunk 2]
```diff
@@ -277,7 +278,26 @@ Waiting for daemon set "multus" rollout to finish: 5 of 6 updated pods are avail
...
+do
+  read -r POD NODE <<< "$i"
+  until oc rsh -n openshift-machine-config-operator "$POD" chroot /rootfs shutdown -r +1
+    do
+      echo "cannot reboot node $NODE, retry" && sleep 3
+    done
+done
+----
+
+** With `ssh`, you can use a bash script similar to the following. The script assumes that you have configured sudo to not prompt for a password.
```
Comment (adellape) [inline modules/nw-ovn-kubernetes-migration.adoc:300]: With the `ssh` command,

File: _topic_maps/_topic_map.yml [hunk 3]
```diff
@@ -1274,6 +1274,10 @@ Topics:
   Topics:
   - Name: About the OpenShift SDN network plugin
     File: about-openshift-sdn
+  - Name: Migrating to the OpenShift SDN network plugin
+    File: migrate-to-openshift-sdn
+  - Name: Rolling back to the OpenShift SDN network plugin
```
Comment (adellape) [inline _topic_maps/_topic_map.yml:1279]: I think this is the only misnamed spot; seems like it should be "Rolling back to the OVN-Kubernetes network plugin".

File: networking/openshift_sdn/rollback-to-ovn-kubernetes.adoc [hunk 4]
```diff
@@ -0,0 +1,13 @@
+:_content-type: ASSEMBLY
+[id="roll-back-to-ovn-kubernetes"]
+= Roll back to the OVN-Kubernetes network plugin
```
Comment (adellape) [inline networking/openshift_sdn/rollback-to-ovn-kubernetes.adoc:3]: Mostly you've got "Rolling" except for here it's just "Roll". Suggest syncing this heading up on "Rolling" as well.

**Review by adellape — COMMENTED** [review]
Clever ifeval/attribute use!

**Pull request merged**